File: lib/screens/profile/data_privacy_screen.dart
import 'package:flutter/material.dart';
import 'package:get/get.dart';
import 'package:google_fonts/google_fonts.dart';
import '../../utils/app_colors.dart';
import '../../utils/app_text.dart';
import '../../utils/app_strings.dart';

class DataPrivacyScreen extends StatelessWidget {
  const DataPrivacyScreen({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: AppColors.background,
      appBar: AppBar(
        backgroundColor: AppColors.background,
        elevation: 0,
        leading: IconButton(
          icon: Icon(Icons.arrow_back_ios_new, color: AppColors.textPrimary),
          onPressed: () => Get.back(),
        ),
        title: Text(AppStrings.dataPrivacy, style: AppText.header2),
        centerTitle: true,
      ),
      body: SingleChildScrollView(
        padding: const EdgeInsets.symmetric(horizontal: 20, vertical: 20),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            _buildSection(
              'Data Collection',
              'We collect only the information necessary to provide our services: '
              'account details, transaction history, and device information for security.',
            ),
            const SizedBox(height: 24),
            _buildSection(
              'Data Usage',
              'Your data is used to process transactions, improve our services, '
              'and comply with legal requirements. We never sell your personal information.',
            ),
            const SizedBox(height: 24),
            _buildSection(
              'Data Security',
              'We use industry-standard encryption to protect your data. '
              'Your account is secured with PIN and optional biometric authentication.',
            ),
            const SizedBox(height: 24),
            _buildSection(
              'Your Rights',
              'You have the right to access, correct, or delete your personal data. '
              'Contact support@rexipay.com for any data-related requests.',
            ),
            const SizedBox(height: 24),
            _buildSection(
              'Cookies & Tracking',
              'We use minimal cookies for app functionality. '
              'We do not track your activity for advertising purposes.',
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildSection(String title, String content) {
    return Container(
      padding: const EdgeInsets.all(20),
      decoration: BoxDecoration(
        color: AppColors.cardBackground,
        borderRadius: BorderRadius.circular(16),
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text(
            title,
            style: GoogleFonts.inter(
              fontSize: 16,
              fontWeight: FontWeight.w700,
              color: AppColors.textPrimary,
            ),
          ),
          const SizedBox(height: 12),
          Text(
            content,
            style: AppText.body2,
          ),
        ],
      ),
    );
  }
}

File: rn/src/screens/DataPrivacyScreen.js
import React from 'react';
import { View, Text, TouchableOpacity, ScrollView, StyleSheet } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { MaterialIcons } from '@expo/vector-icons';
import { useTheme } from '../theme/ThemeContext';

const SECTIONS = [
  {
    title: 'Data Collection',
    content:
      'We collect only the information necessary to provide our services: account details, transaction history, and device information for security.',
  },
  {
    title: 'Data Usage',
    content:
      'Your data is used to process transactions, improve our services, and comply with legal requirements. We never sell your personal information.',
  },
  {
    title: 'Data Security',
    content:
      'We use industry-standard encryption to protect your data. Your account is secured with PIN and optional biometric authentication.',
  },
  {
    title: 'Your Rights',
    content:
      'You have the right to access, correct, or delete your personal data. Contact support@rexipay.com for any data-related requests.',
  },
  {
    title: 'Cookies & Tracking',
    content:
      'We use minimal cookies for app functionality. We do not track your activity for advertising purposes.',
  },
];

export default function DataPrivacyScreen() {
  const { colors } = useTheme();
  const navigation = useNavigation();

  return (
    <View style={[styles.container, { backgroundColor: colors.background }]}>
      <View style={styles.header}>
        <TouchableOpacity onPress={() => navigation.goBack()}>
          <MaterialIcons name="arrow-back-ios" size={20} color={colors.textPrimary} />
        </TouchableOpacity>
        <Text style={[styles.title, { color: colors.textPrimary }]}>Data & Privacy</Text>
        <View style={{ width: 24 }} />
      </View>
      <ScrollView contentContainerStyle={styles.content}>
        {SECTIONS.map((s, i) => (
          <View
            key={i}
            style={[styles.section, { backgroundColor: colors.cardBackground }]}
          >
            <Text style={[styles.sectionTitle, { color: colors.textPrimary }]}>{s.title}</Text>
            <Text style={[styles.sectionContent, { color: colors.textSecondary }]}>
              {s.content}
            </Text>
          </View>
        ))}
      </ScrollView>
    </View>
  );
}

const styles = StyleSheet.create({
  container: { flex: 1 },
  header: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingHorizontal: 20,
    paddingTop: 50,
    paddingBottom: 16,
  },
  title: { fontSize: 18, fontWeight: '700' },
  content: { padding: 20, paddingBottom: 40 },
  section: {
    padding: 20,
    borderRadius: 16,
    marginBottom: 24,
  },
  sectionTitle: { fontSize: 16, fontWeight: '700' },
  sectionContent: { fontSize: 14, marginTop: 12, lineHeight: 22 },
});
